### PR TITLE
etcd-tester: no error for compact double-send

### PR DESCRIPTION
**Pull request by gyuho**
When compactKV request is halted before final acknowledgement,
it used to just continue on the next endpoint. But there could be
a case than compactKV is requested twice, and the first one is already
replicated and applied by the time the second request was to be
applied (returning compact revision error). This skips the case
by parsing the error message.

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @heyitsanthony @xiang90 So I think this fixes the compaction issue, but I just got another hash difference error running this branch.

```
2016-03-30 20:38:44.612767 I | etcd-tester: [round#5 case#0] checking current storage hashes failed (map[10.240.0.2:2379:418588413 10.240.0.3:2379:1358886971 10.240.0.4:2379:1913554223])
```

File: tools/functional-tester/etcd-tester/cluster.go [hunk 1]
```diff
@@ -337,8 +337,11 @@ func (c *cluster) compactKV(rev int64) error {
 		_, err = kvc.Compact(ctx, &pb.CompactionRequest{Revision: rev, Physical: true})
 		cancel()
 		conn.Close()
-		if err == nil {
-			return nil
+		if err != nil {
+			if strings.Contains(err.Error(), "required revision has been compacted") && i > 0 {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/cluster.go]: when does this expect to happen? Why do we send duplicated compaction request at the first place?

Comment (heyitsanthony) [inline tools/functional-tester/etcd-tester/cluster.go]: @xiang90 if the RPC acks on physical compaction, the Compact needs to be sent to every endpoint to be sure it's physically completed on every node

Comment (xiang90) [inline tools/functional-tester/etcd-tester/cluster.go]: oh. ok .reasonable.

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Thanks!

File: tools/functional-tester/etcd-tester/cluster.go [hunk 2]
```diff
@@ -327,18 +327,24 @@ func (c *cluster) compactKV(rev int64) error {
...
+				plog.Printf("%s is already compacted with %d (%v)", u, rev, err)
+				err = nil // in case compact was requested more than once
+			} else {
+				plog.Printf("%s got %v (retrying...)", u, err)
```
Comment (heyitsanthony) [inline tools/functional-tester/etcd-tester/cluster.go]: did it turn out that this retry is necessary?

Comment (gyuho) [inline tools/functional-tester/etcd-tester/cluster.go]: Actually doesn't need this. Never got to see this running. Just removed. Thanks.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
thanks. merging in greens.

**Pull request merged**